docs/changelog.jl

```diff
@@ -0,0 +1,138 @@
+# =================================================================================
+#  Utilities for managing CHANGELOG.md and generating a Documenter-friendly version
+# =================================================================================
```

Comment:
it seems like this file could even be in `src` so it can be re-used by any project using Documenter?

Reply:
I think this should be re-used, yes, but I lean towards this being in a separate package, so that it could be versioned independently and hopefully more rapidly than Documenter itself.

That said, we _could_ still drop it into `src/` if you want to re-use this for e.g. JuMP right away, and just not promise this as official API (but secretly promise to you that we won't drop it on the `1.0.x` branch, for example)?

Reply:
Or, to be fair, if we generally agree that this should be a separate package, we could just copy-paste it into that new package now, and tag that as 0.1.0, and then iterate on that as needed.